RELEASE.md

```diff
@@ -0,0 +1,126 @@
+    - [Making release branch specific changes](#making-release-branch-specific-changes)
+  - [Drafting RCs (Release Candidates)](#drafting-rcs-release-candidates)
+    - [Release Candidate Storage](#release-candidate-storage)
+    - [Cherry Picking Fixes](#cherry-picking-fixes)
+  - [Promoting RCs to Stable](#promoting-rcs-to-stable)
+- [Special Topics](#special-topics)
+  - [Updating submodules for a release](#updating-submodules-for-a-release)
+
+<!-- tocstop -->
+
+## General Overview
+
+Releasing a new version of PyTorch generally entails 3 major steps:
+
+1. Cutting a release branch and making release branch specific changes
+2. Drafting RCs (Release Candidates), and merging cherry picks
+3. Promoting RCs to stable
+
+## Cutting release branches
+
+Release branches are typically cut from the branch [`viable/strict`](https://github.com/pytorch/pytorch/tree/viable/strict) as to ensure that tests are passing on the release branch.
+
+Release branches *should* be prefixed like so:
+```
+release/{MAJOR}.{MINOR}
+```
+
+An example of this would look like:
+```
+release/1.8
+```
+
+Please make sure to create branch that pins divergent point of release branch from the main thunk, i.e. `orig/release/{MAJOR}.{MINOR}`
+### Making release branch specific changes
+
+These are examples of changes that should be made to release branches so that CI can function normally on
+them:
+
+* Update target determinator to use release branch:
+  * Example: https://github.com/pytorch/pytorch/pull/40712
+* Cutting a release branch on [`pytorch/xla`](https://github.com/pytorch/xla)
+  * Example: https://github.com/pytorch/pytorch/pull/40721
+* Update backwards compatibility tests to use RC binaries instead of nightlies
+  * Example: https://github.com/pytorch/pytorch/pull/40706
+
+Create PR from `release/{MAJOR}.{MINOR}` to `orig/release/{MAJOR}.{MINOR}` in order to start CI testing for cherry-picks into release branch.
+
+Example:
+* https://github.com/pytorch/pytorch/pull/51995
+
+ TODO: Create release branch in `builder` repo and pin release CI to use that branch rather than HEAD of builder repo.
+## Drafting RCs (Release Candidates)
+
+To draft RCs, a user with the necessary permissions can push a git tag to the main `pytorch/pytorch` git repository.
+
+The git tag for a release candidate must follow the following format:
+```
+v{MAJOR}.{MINOR}.{PATCH}-rc{RC_NUMBER}
+```
+
+An example of this would look like:
+```
+v1.8.1-rc1
+```
+
+Pushing a release candidate should trigger the `binary_builds` workflow within CircleCI using [`pytorch/pytorch-probot`](https://github.com/pytorch/pytorch-probot)'s [`trigger-circleci-workflows`](trigger-circleci-workflows) functionality.
+
+This trigger functionality is configured here: [`pytorch-circleci-labels.yml`](https://github.com/pytorch/pytorch/blob/master/.github/pytorch-circleci-labels.yml)
+
+### Release Candidate Storage
+
+Release candidates are currently stored in the following places:
```

Comment:
How does an RC go from a CircleCI artifact to S3?

Reply:
Those are handled through this script: https://github.com/pytorch/pytorch/blob/master/.circleci/scripts/binary_upload.sh